crates/wasmtime/src/compile/runtime.rs

```diff
@@ -137,7 +137,7 @@ impl FinishedObject for MmapVecWrapper {
 
             fn reserve(&mut self, additional: usize) -> Result<(), ()> {
                 assert!(self.mmap.is_none(), "cannot reserve twice");
-                self.mmap = match MmapVec::with_capacity(additional) {
+                self.mmap = match MmapVec::with_capacity_and_alignment(additional, 1) {
```

Comment:
I think this may not work for the use case of combining `CustomCodeMemory` and compilation at runtime, so the `Engine` and its custom alignment might need to be plumbed down here?

Reply:
Done -- this was slightly gnarly to plumb through (required an extra `State` arg on the `FinishedObject` trait in `environ` and propagated through all the delegated builder stuff and cache handling and ...) but seems to work.